// angular2-flash-messages/__ivy_ngcc__/module/module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var flash_messages_component_1 = require("./flash-messages.component");
var flash_messages_service_1 = require("./flash-messages.service");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./flash-messages.component');
var ɵngcc2 = require('@angular/common');
var FlashMessagesModule = /** @class */ (function () {
    function FlashMessagesModule() {
    }
    FlashMessagesModule.forRoot = function () {
        return {
            ngModule: FlashMessagesModule,
            providers: [flash_messages_service_1.FlashMessagesService]
        };
    };
FlashMessagesModule.ɵfac = function FlashMessagesModule_Factory(t) { return new (t || FlashMessagesModule)(); };
FlashMessagesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FlashMessagesModule });
FlashMessagesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ providers: [], imports: [[common_1.CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FlashMessagesModule, { declarations: [ɵngcc1.FlashMessagesComponent], imports: [ɵngcc2.CommonModule], exports: [ɵngcc1.FlashMessagesComponent] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FlashMessagesModule, [{
        type: core_1.NgModule,
        args: [{
                imports: [common_1.CommonModule],
                declarations: [flash_messages_component_1.FlashMessagesComponent],
                exports: [flash_messages_component_1.FlashMessagesComponent],
                providers: []
            }]
    }], function () { return []; }, null); })();
    return FlashMessagesModule;
}());
exports.FlashMessagesModule = FlashMessagesModule;
